File: P2.cc
//fibonacci sequence

#include <iostream>

int main()
{

  long N_max = 4e6;
  long n1 = 1;
  long n2 = 2;
  long n = 0;

  long long Sum = n2;

  while (n < N_max)
    {

      n = n1 + n2;

      if (n > N_max)
	{
	  break;
	}
      else
	{
	  n1 = n2;
	  n2 = n;

	  if (n % 2 == 0)     
	    {	  
	      Sum += n;
	    }
	}
    }

  std::cout << Sum << "\n";

  return 0;

}

File: P2.py
#!/usr/bin/env python
"""Even Fibonacci numbers"""

from P25 import FibonacciNumbers

MAX_VALUE = int(4e6)


def main():
    """Sum even Fibonacci terms below 4 million."""
    sum_terms = 0
    for term in FibonacciNumbers(1, 2):
        if term.value <= MAX_VALUE:
            if term.value % 2 == 0:
                sum_terms += term.value
        else:
            break
    print sum_terms


if __name__ == '__main__':
    main()
